sharding/node/utils.go

```diff
@@ -0,0 +1,64 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/sharding"
+	"github.com/ethereum/go-ethereum/sharding/contracts"
+)
+
+// dialRPC endpoint to node.
+func dialRPC(endpoint string) (*rpc.Client, error) {
+	if endpoint == "" {
+		endpoint = node.DefaultIPCEndpoint(clientIdentifier)
+	}
+	return rpc.Dial(endpoint)
+}
+
+// initSMC initializes the sharding manager contract bindings.
+// If the SMC does not exist, it will be deployed.
+func initSMC(n *shardingNode) (*contracts.SMC, error) {
+	b, err := n.client.CodeAt(context.Background(), sharding.ShardingManagerAddress, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get contract code at %s: %v", sharding.ShardingManagerAddress, err)
+	}
+
+	// Deploy SMC for development only.
+	// TODO: Separate contract deployment from the sharding node. It would only need to be deployed
+	// once on the mainnet, so this code would not need to ship with the node.
+	if len(b) == 0 {
+		log.Info(fmt.Sprintf("No sharding manager contract found at %s. Deploying new contract.", sharding.ShardingManagerAddress.String()))
+
+		txOps, err := n.CreateTXOpts(big.NewInt(0))
+		if err != nil {
+			return nil, fmt.Errorf("unable to intiate the transaction: %v", err)
+		}
+
+		addr, tx, contract, err := contracts.DeploySMC(txOps, n.client)
+		if err != nil {
+			return nil, fmt.Errorf("unable to deploy sharding manager contract: %v", err)
+		}
+
+		for pending := true; pending; _, pending, err = n.client.TransactionByHash(context.Background(), tx.Hash()) {
```

Comment:
No need for the pending conditional to make a while loop
```go
		for {
			if err = n.client.TransactionByHash(context.Background(), tx.Hash()); err != nil {
				return nil, fmt.Errorf("unable to get transaction by hash: %v", err)
			}
			time.Sleep(1 * time.Second)
		}
```

Reply:
Getting: unreachable code right below it

Reply:
Ok, I misunderstood the code here.